src/service.rs

```diff
@@ -1188,10 +1194,20 @@ where
             }
             ServiceTask::ProtocolOpen { session_id, target } => match target {
                 TargetProtocol::All => {
-                    let ids = self.protocol_configs.keys().copied().collect::<Vec<_>>();
-                    ids.into_iter().for_each(|id| {
-                        self.protocol_open(cx, session_id, id, String::default(), Source::External)
-                    });
+                    // Borrowed check attack
+                    #[allow(clippy::needless_collect)]
```

Comment:
Why do we need to introduce a new `allow clippy` here, it looks like the PR didn't upgrade the rust version.

Reply:
Most of the clippy modifications here are the newly added warnings of clippy version 1.47, and are forward compatible, I think they can be added